pkg/cmd/root/root.go

```diff
@@ -136,6 +164,51 @@ func NewCmdRoot(f *cmdutil.Factory, version, buildDate string) *cobra.Command {
 		}
 	}
 
+	// Aliases
+	aliases := cfg.Aliases()
+	for k, v := range aliases.All() {
+		aliasName := k
+		aliasValue := v
+		isShellAlias := strings.HasPrefix(aliasValue, "!")
+		split, _ := shlex.Split(aliasName)
+		parentCmd, parentArgs, _ := cmd.Find(split)
+		// If there are no parent args the user defined an alias that shadows an existing command.
+		if parentCmd != nil && len(parentArgs) == 1 {
+			if !parentCmd.ContainsGroup("alias") {
+				parentCmd.AddGroup(&cobra.Group{
+					ID:    "alias",
+					Title: "Alias commands",
+				})
+			}
+			if isShellAlias {
+				shellAliasCmd := NewCmdShellAlias(io, parentArgs[0], aliasValue)
+				parentCmd.AddCommand(shellAliasCmd)
+				parentCmd.ValidArgs = append(parentCmd.ValidArgs, fmt.Sprintf("%s\tShell alias", aliasName))
+			} else {
+				aliasCmd := NewCmdAlias(io, parentArgs[0], aliasValue)
+				split, _ := shlex.Split(aliasValue)
+				child, _, _ := cmd.Find(split)
+				aliasCmd.SetUsageFunc(func(_ *cobra.Command) error {
+					return rootUsageFunc(f.IOStreams.ErrOut, child)
+				})
+				aliasCmd.SetHelpFunc(func(_ *cobra.Command, args []string) {
+					rootHelpFunc(f, child, args)
+				})
```

Comment:
This is to maintain status quo with how current aliases work where the help and usage output is actually the help and usage for the targeted command.